_posts/2020-09-10-rustpython-usecase.md

```diff
@@ -0,0 +1,59 @@
+
+Usually re-thinking how something works requires thinking with a long-term mindset, being open to experimentation and trying a new approach instead of adopting incremental improvements. Thinking outside the box is easier said than done. Here is where RustPython comes in.
+
+## The role of RustPython
+
+The promise of RustPython for scientific computing in the browser is simple. Rust is a new and modern programming language. It plays nicely with WebAssembly. By writing an interpreter for Python in Rust, you can compile that interpreter to WebAssembly. Since WebAssembly runs in the browser, this means you get yourself an interpreter for Python that runs in the browser. In other words, someone can type python code in a browser and the browser can load an interpreter that understands it.
+
+There is a merit to a fresh start with a new Python interpreter that has "running in the browser" as a use case. There is freedom of choice that comes from not having to worry about a huge codebase and a benefit to using a modern programming language like Rust.
+
+The trade-off is that you have a whole interpreter to write, a standard library and so many other libraries to support. It is not a road for the faint hearted or for someone who needs a production ready environment. In fact, while RustPython supports the full Python syntax, it doesn't support the Standard Library at the moment.
+
+## What about?
+When talking about new tools, technical debate can get intense -- very quickly. Sure, there are a few other approaches, For example:
+- What about compiling CPython to WebAssembly?
+- What about using Javascript for scientific computing? Javascript is very advanced and much different from what it used to be!
+- What's wrong with servers and what about using Docker?
+- What about someone just paying for a lot of servers for a lot of scientists?
+    
+All the above approaches have their merit. Usually, there are multiple ways to solve a problem. Each approach brings together a different community with different mindsets, toolsets and skillsets. Working on one solution rarely takes away from the viability of another. In the occasion when the community overlaps, for example, a programmer who knows C and Rust can choose to contribute to whichever interpreter they want. That decision will have to do with their preference, the community and what he/she think has a higher chance of working. More people and more energy to solve a problem means a bigger diversity in solutions. One of these solutions is meant to work or inspire and lead to another. Competition and diversity as economists and social scientists believe, is good.
+
+Besides, scientific computing in the browser is one of the use cases for RustPython. Another use case for example is exposing Python scripting in an application written in Rust. This would allow quickly tweaking logic where Rust's compile times would be inhibitive.
```

Comment:
```suggestion
Besides, scientific computing in the browser is only one of the use cases for RustPython; it can also used to embed python scripting in a rust program or as an alternate interpreter that supports threading without the GIL.
```
You don't need to go super into depth into alternate use cases, cause this is mainly focused on scientific computing in the browser.